odata-protocol/11.4 Data Modification.md

```diff
@@ -1078,18 +1086,29 @@ If an individual change fails due to a failed dependency, it MUST be
 annotated with the term [`Core.DataModificationException`](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Core.V1.md#DataModificationException) and SHOULD specify
 a `responseCode` of `424` ([Failed Dependency](#ResponseCode424FailedDependency)).
 
-Alternatively, the verb `PUT` can be used, in which case the request
-body MUST be the representation of a collection of entities. In this
+### ##subsubsec Replace a Collection of Entities
+Collections of entities can be replaced by submitting a `PUT` request
+to the resource path of the collection. The body of the request MUST be
+the representation of the complete collection of replacement entities. In this
 case all entities provided in the request are applied as
 [upserts](#UpsertanEntity), and any entities not provided in the request
-are deleted. In this case, if the `continue-on-error` preference has
-been specified, and the request returns a success response code, then a
-response MUST be returned regardless of the
+are deleted. 
+
+For each entity being updated, clients MAY specify an [ETag](#UseofETagsforAvoidingUpdateConflicts)
+value obtained from a previous request. If an ETag is provided that does not match the ETag 
+value of the entity being updated, or if an ETag is provided for an entity that does not 
+currently exist, then services that support ETags MUST NOT apply the 
+change and instead report a `412 Precondition Failed` error according to the appropriate [`continue-on-error`](#ErrorHandlingwhenReplacingaCollectionofEntities) behavior. 
+The special ETag value `*` can be used to match any existing entity but fail if the entity does not already exist.
+
+#### ##subsubsubsec Error Handling when Replacing a Collection of Entities
```

Comment:
This section lacks the changes I suggested with https://github.com/oasis-tcs/odata-specs/pull/1958#discussion_r1648484613.

Reply:
Added it to section "Update a Collection of Entities".

Reply:
Status code 412 must also be mentioned for the case where the `continue-on-error` preference has not been applied.

Reply:
With "stop on first error" the rules in the referenced section on "Upsert an Entity" apply, which in turn references sections "Create an Entity" and "Update an Entity". These sections already describe the If-Match behavior.

Reply:
By stating "according to the appropriate `continue-on-error` behavior, my intent was to capture both continue-on-error = true (don't make the change to the record and return 412 for the failed record) or continue-on-error = false (don't make any changes and return 412 for the response header, as outlined in the if-not-match section).

That is, the error for not matching is 412 Precondition Failed. Just like any other error that occurs for an individual entity, where the error is returned depends on continue-on-error. If True, then it's returned for the entity (just like any other error that prevented an individual update would be returned), if False then it's returned for the response (just like any other individual error that resulted in the whole operation failing due to continue-on-error=false would be returned as the response code for the whole operation).  There's nothing special about handling a 412, versus any other error that might prevent an individual entity from being updated.